src/marqo/s2_inference/clip_utils.py

```diff
@@ -156,15 +185,195 @@ def __init__(self, model_type: str = "ViT-B/32", device: str = 'cpu',  embedding
         self.processor = None
         self.embedding_dimension = embedding_dim
         self.truncate = truncate
+        self.model_properties = kwargs["model_properties"]
 
     def load(self) -> None:
 
-        # https://github.com/openai/CLIP/issues/30
-        self.model, self.preprocess = clip.load(self.model_type, device='cpu', jit=False)
-        self.model = self.model.to(self.device)
-        self.tokenizer = clip.tokenize
-        self.model.eval()
-    
+        try:
+            # The original method to load the openai clip model
+            # https://github.com/openai/CLIP/issues/30
+            self.model, self.preprocess = clip.load(self.model_type, device='cpu', jit=False)
+            self.model = self.model.to(self.device)
+            self.tokenizer = clip.tokenize
+
+
+        except RuntimeError:
+            logger.info("Can not load clip model. Try custom clip model loading.")
+
+            path = self.model_properties.get("localpath", None) or self.model_properties.get("url", None)
+            if path is None:
+                raise InvalidModelPropertiesError("Model path `url` or `localpath` is not provided. Can not load generic model.")
+            elif os.path.isfile(path):
+                self.model_path = path
+            elif validators.url(path):
+                self.model_path = download_pretrained_from_url(path)
+
+            self.jit = self.model_properties.get("jit", False)
+            self.device = self.model_properties.get("device", "cpu")
+            self.mean = self.model_properties.get("mean", None)
+            self.std = self.model_properties.get("std", None)
+
+
+            try:
```

Comment:
is there another way to determine if the model belongs to CLIP or open_clip instead of the try/except?